src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -0,0 +1,128 @@
+import subprocess
+from typing import TYPE_CHECKING, Union
+
+import numpy as np
+
+from ..utils import logging
+from .base import Pipeline
+
+
+if TYPE_CHECKING:
+    from ..models.auto import AutoFeatureExtractor
+
+logger = logging.get_logger(__name__)
+
+
+def ffmpeg_read(bpayload: bytes, sampling_rate: int) -> np.array:
```

Comment:
Not too sure about this function, but if we enable it we should have it gracefully fail if the dependency is not found on the system

Reply:
I'm not sure either. 

We already seem to use `soundfile` but `soundfile` won't read as many files + will fail if the sampling rate is incorrect. It feels pretty bad UX for pipeline users.

Reply:
How about `librosa.load(path, sr=sampling_rate, mono=True)`? https://librosa.org/doc/latest/generated/librosa.load.html#librosa.load
It uses `soundfile` as a default backend and falls back on `audioread`+`ffmpeg` for unsupported codecs like mp3.
Also, loading and resampling+mono in a single line is pretty nice :)

Reply:
`librosa` is not a dependency currently, I'm more than hesitant to add it for just that feature.

Also `librosa` api enforces to use a filename which is pretty bad in the context of an API (where hitting disk is never desirable). It does fallback to ffmpeg anyway, in pretty much what `ffmpeg_read` does. (Actually if you use soundfile and resample later then you're also wasting a lot of time, it's much better if the file can be read at the correct sampling rate directly).

The main issue I think is : should we have such functions in `transformers` ? 
It makes life easier on users which is Priority number 1 of the lib (especially pipelines).
At the same time, I don't think we want to reinvent all image/audio preprocessing.

Note: `torchaudio` is another alternative to `librosa` to consider too. (it doesn't do automatic resampling afaik)

Reply:
@mpariente and @jonashaag might also want to chime in on this because I remember having discussions on this subject with them

Reply:
The question is what is your priority.

- If you are looking to minimise dependencies while being able to read a lot of file types, this way is probably best. (Could switch the manual ffmpeg invocation for the `audioread` library.)
- If you are looking to minimise load time, then trying soundfile + [julius](https://github.com/adefossez/julius) is probably the best choice. (Could also add a fallback to ffmpeg.)
- To keep things as simple as possible and to write is little code as possible, just use `librosa.load` which tries soundfile first and then falls back to ffmpeg.

> (Actually if you use soundfile and resample later then you're also wasting a lot of time, it's much better if the file can be read at the correct sampling rate directly).

I don't think this is correct since ffmpeg also needs to do resampling, so it depends on the speed of ffmpeg resampling implementation vs. your resampling implementation.

Reply:
We should build a library for reading audio files in Rust :-)

Reply:
> We should build a library for reading audio files in Rust with Python bindings

Famous last words

Reply:
Thanks so much @jonashaag! Am I wrong in remembering that `julius`' resampling was more or less merged into recent versions of torchaudio? (also cc'ing @adefossez)

Reply:
Yes it was!

Reply:
@LysandreJik do you have any opinions ? 

I'll vote in favor of this current implementation, it's pretty small code-wise, is probably the largest audio support of the proposed alternatives.

Reply:
Ok for me, let's go with the current proposition.

Reply:
I'm joining a bit late the conversation now that this is merged. ffmpeg is great but a bit of a heavy dependency and also it is not possible to ensure which format it will support, depending on how it was installed.
Also, ffmpeg cannot be installed automatically as a dependency through pip, or requirements in a `setup.py` file. 

For the latest version of Demucs, I ended up doing a combination of torchaudio (doing resampling with `julius` and manual channel conversion) + fallback to ffmpeg if torchaudio failed, because torchaudio on Windows has a really limited support (no mp3, although that will change soon), so that on Linux and Mac it will work with a simple `pip install`, while on Windows, I still need people to use Anaconda to install ffmpeg.

https://github.com/facebookresearch/demucs/blob/master/demucs/separate.py#L21 (in there it first tried torchaudio, then fallback to `AudioFile` which is a similar wrapper around ffmpeg).

Reply:
thanks @adefossez